packages/runtime/src/lib/itxn.ts

```diff
@@ -65,9 +65,8 @@ export function setInnerTxField (
   }
 
   if (addrTxnFields.has(field)) {
-    const assertedVal = op.assertBytes(val, line);
-    const accountState = interpreter.getAccount(assertedVal, line);
-    txValue = Buffer.from(decodeAddress(accountState.address).publicKey);
+    const assertedAddr = op.assertAddress(val, line);
+    txValue = assertedAddr;
```

Comment:
```suggestion
    txValue = op.assertAddress(val, line);
```

Reply:
This is not entirely correct. We only want to remove `interpreter.getAccount` check only in the case of "acfg" transaction. So for example if i do an asset transfer transaction from smart contract's account to `<addr>`, then that "<addr>" must be present in the accounts array (passed in first txn account).

Reply:
Also, we can modify `interpreter.getAccount` function here. The address must be present in the array (if txtype is not "acfg"), but it doesn't have to be an already initialized runtime account

Reply:
> This is not entirely correct. We only want to remove `interpreter.getAccount` check only in the case of "acfg" transaction. So for example if i do an asset transfer transaction from smart contract's account to `<addr>`, then that "" must be present in the accounts array (passed in first txn account).

Teal allow we declare `itxn_field TypeEnum` before and after `itxn_field <address>` and Teal only wants to know this is the correct address or not.